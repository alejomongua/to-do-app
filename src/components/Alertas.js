import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { removeAlert } from '../redux/actions'

const TIMEOUT_ALERTAS = 3000

class Alertas extends Component {
  render() {
    return (
      <div>
        {
          this.props.alertas.map(alerta => {
            const clases = alerta.type === 'error'
              ? 'bg-red-100 border-red-400 text-red-700'
              : 'bg-green-100 border-green-400 text-green-700'

            setTimeout(() => this.props.removeAlert(alerta), TIMEOUT_ALERTAS)
            
            return (
              <div key={alerta.text} className={`${clases} border px-4 py-3 rounded relative`} role='alert'>
                {alerta.text}
              </div>
            )
          })
        }
      </div>
    )
  }
}

Alertas.propTypes = {
  alertas: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      type: PropTypes.string,
    })
  )
}

const mapStateToProps = (state) => ({
  alertas: state.alertas
})

const mapDispatchToProps = (dispatch) => ({
  removeAlert (alert) {
    dispatch(removeAlert(alert))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Alertas)
