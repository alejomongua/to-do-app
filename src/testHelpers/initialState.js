export default {
  alertas: [],
  todos: {
    list: ['0', '1', '2', '3', '4'],
    byId: {
      '0': {
        text: 'Tarea 1',
        completed: false,
      },
      '1': {
        text: 'Tarea 2',
        completed: true,
      },
      '2': {
        text: 'Tarea 3',
        completed: false,
      },
      '3': {
        text: 'Tarea 4',
        completed: true,
      },
      '4': {
        text: 'Tarea 5',
        completed: false,
      },
    }
  },
  visibilityFilter: 'SHOW_ALL',
}
