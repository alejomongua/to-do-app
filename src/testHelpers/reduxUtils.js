// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import reducer from '../redux/reducers'
import initialState from '../testHelpers/initialState'

const create = renderer.create

delete renderer.create

renderer.create = function (
  ui
) {
  return create(<Provider store={createStore(reducer, initialState)}>{ui}</Provider>)
}

function render(
  ui
) {
  function Wrapper({ children }) {
    return <Provider store={createStore(reducer, initialState)}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
export { renderer }