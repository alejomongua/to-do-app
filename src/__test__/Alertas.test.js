import '@babel/polyfill'
import React from 'react';
import Alertas from '../components/Alertas'
import { render, renderer, screen } from '../testHelpers/reduxUtils'
import _initialState from '../testHelpers/initialState'
import '@testing-library/jest-dom'

it("renders without crashing", () => {
  render(<Alertas />)
})

it('matches snapshot', () => {
  const tree = renderer
    .create(<Alertas />)
    .toJSON();
  expect(tree).toMatchSnapshot()
})

it('shows alerts', () => {
  const textoAlerta = 'Esto es un mensaje de alerta'
  const initialState = { ..._initialState }
  initialState.alertas = [{
    type: 'error',
    text: textoAlerta
  }]

  render(<Alertas />, { initialState })

  screen.debug()

  const tree = renderer
    .create(<Alertas />, { initialState })
    .toJSON();
  expect(tree).toMatchSnapshot()

  expect(screen.getByText(textoAlerta)).toBeInTheDocument()
})