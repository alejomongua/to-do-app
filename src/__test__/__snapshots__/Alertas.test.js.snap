// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<div />`;

exports[`shows alerts 1`] = `
<div>
  <div
    className="bg-red-100 border-red-400 text-red-700 border px-4 py-3 rounded relative"
    role="alert"
  >
    Esto es un mensaje de alerta
  </div>
</div>
`;
